### PR TITLE
Fix Smartgunner farsight

### PR DESCRIPTION
**Pull request by Vermidia**
## About the PR

Fix smartgunner farsight zoom staying when unequipped. It also won't work or give the action outside of the eye slot now.

## Why / Balance

Discord issues, you could give a whole squad farsight with this.

## Technical details

Added equipped and unequipped events to check for if it's been removed or re-added (with farsight on). When reseting zoom, the EyeComponent is now gotten to reset max zoom to whatever zoom is defined there. Also gave the farsight component a slotflags field so it has to be in certain slots to count as equipped.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/user-attachments/assets/d8463e72-ea74-450f-93bc-348f3432dc9e

(I recorded this before fixing it not working when equipped and already active)

## Breaking changes

**Changelog**

:cl:
- fix: Smartgunner combat-sight giving permanent far sight

### CODE REVIEW

**Comment by Dutch-VanDerLinde** [comment]
does this fix runner farsight glitch too?

**Comment by Vermidia** [comment]
Different components, so no I think

File: Content.Shared/_RMC14/FarSight/FarSightSystem.cs [hunk 1]
```diff
@@ -42,10 +48,47 @@ private void OnFarSightAction(Entity<FarSightItemComponent> ent, ref FarSightAct
...
+            _eye.SetMaxZoom(user, ent.Comp.Zoom);
+            _eye.SetZoom(user, ent.Comp.Zoom);
+        }
+        else
+        {
+            if (TryComp<EyeComponent>(user, out var eye))
+                _eye.SetMaxZoom(user, eye.Zoom);
+
+            _eye.ResetZoom(user);
+        }
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/FarSight/FarSightSystem.cs:79]: This should be its own method so that this and OnFarSightAction don't have to repeat it

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**